pageserver/src/http/routes.rs

```diff
@@ -2424,6 +2426,102 @@ async fn list_aux_files(
     json_response(StatusCode::OK, files)
 }
 
+async fn perf_info(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_shard_id: TenantShardId = parse_request_param(&request, "tenant_shard_id")?;
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+    check_permission(&request, Some(tenant_shard_id.tenant_id))?;
+
+    let state = get_state(&request);
+
+    let timeline =
+        active_timeline_of_active_tenant(&state.tenant_manager, tenant_shard_id, timeline_id)
+            .await?;
+
+    // First, collect all split points of the layers.
+    let mut split_points = BTreeSet::new();
+    let mut delta_ranges = Vec::new();
+    let mut image_ranges = Vec::new();
+
+    let all_layer_files = {
+        let guard = timeline.layers.read().await;
+        guard.all_persistent_layers()
+    };
+    let lsn = timeline.get_last_record_lsn();
+
+    for key in all_layer_files {
+        if key.is_delta {
+            split_points.insert(key.key_range.start);
+            split_points.insert(key.key_range.end);
+            delta_ranges.push((key.key_range.clone(), key.lsn_range.clone()));
+        } else {
+            split_points.insert(key.key_range.start);
+            split_points.insert(key.key_range.end);
+            image_ranges.push((key.key_range.clone(), key.lsn_range.start));
+        }
+    }
+
+    // For each split range, compute the estimated read amplification.
+    let split_points = split_points.into_iter().collect::<Vec<_>>();
+
+    #[derive(serde::Serialize)]
+    struct RangeAnalysis {
+        start: String,
+        end: String,
+        has_image: bool,
+        num_of_deltas_above_image: usize,
+        total_num_of_deltas: usize,
+    }
+
+    let mut result = Vec::new();
+
+    for i in 0..(split_points.len() - 1) {
+        let start = split_points[i];
+        let end = split_points[i + 1];
+        // Find the latest image layer that contains the information
+        let mut maybe_image_layers = image_ranges
+            .iter()
+            .filter(|(key_range, img_lsn)| key_range.contains(&start) && img_lsn <= &lsn)
+            .cloned()
+            .collect::<Vec<_>>();
+        maybe_image_layers.sort_by(|a, b| a.1.cmp(&b.1));
+        let image_layer = maybe_image_layers.last().cloned();
```

Comment:
This doesn't check that the whole `[start, end)` key range of a delta layer is covered by the `image_layer`.

Ah, it doesn't need to check that because we insert split points for image layers also.

Maybe worth a comment?